dispatch/apps/api/views.py

```diff
@@ -401,16 +403,33 @@ class EventViewSet(DispatchModelViewSet):
     serializer_class = EventSerializer
 
     def get_queryset(self):
-        queryset = Event.objects.all()
+        # Only authenticated users can see submission or unpublished events
+        if self.request.user.is_authenticated():
+            queryset = Event.objects.all()
+        else:
+            queryset = Event.objects.filter(
+                Q(is_submission__exact=False),
```

Comment:
I don't think you need to use `__exact` here -- `is_submission=False` should be enough